packages/teams-js/src/public/pages.ts

```diff
@@ -752,4 +752,91 @@ export namespace pages {
       return runtime.supports.pages ? (runtime.supports.pages.appButton ? true : false) : false;
     }
   }
+
+  /**
+   * Provides functions for navigating without needing to specify your application ID.
+   *
+   * @beta
+   */
+  export namespace currentApp {
+    /**
+     * Parameters for the NavigateWithinApp
+     *
+     * @beta
+     */
+    export interface NavigateWithinAppParams {
+      /**
+       * As of 2.0.0, please use {@link app.PageInfo.id | app.Context.page.id} instead
+       *
+       * Developer-defined ID of the Page to navigate to within the application (Formerly EntityID)
+       */
+      pageId: string;
+
+      /**
+       * As of 2.0.0, please use {@link app.PageInfo.subPageId | app.Context.page.subPageId} instead
+       *
+       * Optional developer-defined ID describing the content to navigate to within the Page. This will be passed
+       * back to the application via the Context object.
+       */
+      subPageId?: string;
+    }
+
+    /**
+     * Navigate within the currently running application with page ID, and sub-page ID (for navigating to
+     * specific content within the page).
+     * @param params - Parameters for the navigation
+     * @returns a promise that will resolve if the navigation was successful
+     *
+     * @beta
+     */
+    export function navigateTo(params: NavigateWithinAppParams): Promise<void> {
+      return new Promise<void>((resolve) => {
+        ensureInitialized(
+          FrameContexts.content,
+          FrameContexts.sidePanel,
+          FrameContexts.settings,
+          FrameContexts.task,
+          FrameContexts.stage,
+          FrameContexts.meetingStage,
+        );
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
+        resolve(send('pages.currentApp.navigateTo', params));
+      });
+    }
+
+    /**
+     * Navigate to the currently running application's first static page defined in the application
+     * manifest.
+     * @beta
+     */
+    export function navigateToDefaultPage(): Promise<void> {
+      return new Promise<void>((resolve) => {
+        ensureInitialized(
+          FrameContexts.content,
+          FrameContexts.sidePanel,
+          FrameContexts.settings,
+          FrameContexts.task,
+          FrameContexts.stage,
+          FrameContexts.meetingStage,
+        );
+        if (!isSupported()) {
+          throw errorNotSupportedOnPlatform;
+        }
+        resolve(send('pages.currentApp.navigateToDefaultPage'));
```

Comment:
Do we need a separate message to be passed around, or should both these new APIs share a single over-the-wire message, with `pageId` being optional?

Reply:
Having 2 different message strings helps us to understand the code better and isolate the functions.